src/app/ReadHandler.cpp

```diff
@@ -652,6 +719,52 @@ CHIP_ERROR ReadHandler::ProcessSubscribeRequest(System::PacketBufferHandle && aP
 
     mExchangeCtx->WillSendMessage();
 
+#if CHIP_CONFIG_PERSIST_SUBSCRIPTIONS
+    auto * subscriptionResumptionStorage = InteractionModelEngine::GetInstance()->GetSubscriptionResumptionStorage();
+    if (!subscriptionResumptionStorage)
+    {
+        return CHIP_NO_ERROR;
+    }
```

Comment:
ReturnErrorCodeIf(!subscriptionResumptionStorage, CHIP_NO_ERROR).

I believe this makes it important to split in a separate mthod: this early exit only cares about persistent subscriptions, but the ProcessSubscribeRequest is doing more than that. Early exit here (with success!) may be unexpected and is a footgun for future maintenance if more code is added at the end of this method.

Reply:
I've split this part into a separate method.